src/main/java/seedu/address/logic/parser/DisplayDriverTasksCommandParser.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+
+import javafx.collections.ObservableList;
+
+import seedu.address.logic.commands.DisplayDriverTasksCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.person.Driver;
+import seedu.address.model.person.DriverIsTheSamePredicate;
+
+/**
+ * Parses input arguments and creates a new DisplayDriverTasksCommand object
+ */
+public class DisplayDriverTasksCommandParser implements Parser<DisplayDriverTasksCommand> {
+
+    private static ObservableList<Driver> driverList;
+
+    public void setDriverList(ObservableList<Driver> driverList) {
+        this.driverList = driverList;
+    }
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the DisplayDriverTasksCommand
+     * and returns a DisplayDriverTaskCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public DisplayDriverTasksCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DisplayDriverTasksCommand.MESSAGE_USAGE));
+        }
+
+        int driverIndex = Integer.parseInt(trimmedArgs);
+        Driver driver = driverList.get(driverIndex - 1);
```

Comment:
Integer parseInt will throw error if user typed string here.
if ur parse for id, u should use parseId under ParserUtil instead.

and this class is called Parser, so do only  parse id here.
do all the logic stuffs at DisplayDriverTaskCommand instead such as get from driverlist

Reply:
The thing is since i need to pass a driver into the predicate from this class, 
i need to get the driver already. Plus in a sense we can be thinking of this as parsing the argument into a driver right